src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UpdateContactDetailsContestedMidHandler.java

```diff
@@ -38,7 +48,46 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
             caseDetails.getId());
 
         FinremCaseData caseData = caseDetails.getData();
+        List<String> errors = new ArrayList<>();
+        errors.addAll(postalService.validate(caseData));
+        errors.addAll(validateCaseData(caseData));
         return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
-            .data(caseData).errors(postalService.validate(caseData)).build();
+            .data(caseData).errors(errors).build();
+    }
+
+    private List<String> validateCaseData(FinremCaseData caseData) {
+
+        List<String> errors = new ArrayList<>();
+        ContactDetailsWrapper wrapper = caseData.getContactDetailsWrapper();
+
+        if (caseData.isApplicantRepresentedByASolicitor()) {
+            if (wrapper.getApplicantSolicitorAddress() != null
```

Comment:
Just a thought on this solution. 

If the solicitor's postcode is provided, the backend will continue to the next check, which is the applicant's address. If the applicant's postcode is missing, an error may be raised prematurely, even though we haven't navigated to the applicant details page yet due to checking each field sequentially. Is this correct?